rasa/core/trackers.py

```diff
@@ -420,11 +432,67 @@ def undo_till_previous(event_type, done_events):
                 # the `action_listen` action right before it).
                 undo_till_previous(UserUttered, applied_events)
                 undo_till_previous(ActionExecuted, applied_events)
+            elif (
+                isinstance(event, ActionExecuted)
+                and event.action_name in form_names
+                and not self._upcoming_rejection(idx)
```

Comment:
the action that got immediately rejected is not added to the `tracker.events` (unless you changed that, so you're looking for rejection for different action

Reply:
Ah, I didn't know that. Will fix it 👍